test/commands/positive.test.ts

```diff
@@ -82,7 +119,157 @@ describe('Positive', () => {
+        guide: 'gollum',
+      });
+      expect(mockValidateParameters).toHaveBeenCalledWith('getHobbit', {
+        name: 'Frodo',
+      });
+      expect(mockValidateParameters).toHaveBeenCalledWith('getTomBombadil', {
+        times: 2,
+      });
+    });
+
+    describe('parameter validation fails', () => {
+      it('outputs a failure for that operation', async () => {
+        mockValidateParameters.mockImplementation(
+          (operationId) =>
+            new Promise((resolve) => {
+              if (operationId === 'walkIntoMordor')
+                throw new TypeMismatchError(
+                  ['parameters', 'guide', 'example'],
+                  'string',
+                  'number',
+                );
+
+              resolve();
+            }),
+        );
+
+        await expect(async () => {
+          await Positive.run(['http://urldoesnotmatter.com']);
+        }).rejects.toThrow('1 operation failed');
+
+        expect(result).toEqual([
+          'walkIntoMordor: Failed - Actual type did not match schema. Path: parameters -> guide -> example. Schema type: string. Actual type: number\n',
+          'getHobbit: Succeeded\n',
+          'getTomBombadil: Succeeded\n',
+        ]);
+      });
+    });
+
     describe('operation has parameter groups', () => {
+      beforeEach(() => {
+        mockGetParameters.mockImplementation(
+          () =>
+            new Promise((resolve) =>
+              resolve({
+                walkIntoMordor: [
+                  {
+                    door: {
+                      door: 'front',
+                    },
+                  },
+                  {
+                    guided: {
+                      guide: 'gollum',
+                    },
+                  },
+                ],
+                getHobbit: {
+                  default: { name: 'Frodo' },
+                },
+                getTomBombadil: {
+                  default: { times: 2 },
+                },
+              }),
+            ),
+        );
+      });
+
+      describe('one of the parameter groups fails parameter validation', () => {
+        it('does not execute a request for that parameter group', async () => {
+          mockValidateParameters.mockImplementation(
+            (operationId, wrappedParameters) =>
+              new Promise((resolve) => {
+                if (
+                  operationId === 'walkIntoMordor' &&
+                  Object.keys(wrappedParameters)[0] === 'guided'
+                )
+                  throw new TypeMismatchError(
+                    ['parameters', 'guide', 'example'],
+                    'string',
+                    'number',
+                  );
+
+                resolve();
+              }),
+          );
+          await expect(async () => {
+            await Positive.run(['http://urldoesnotmatter.com']);
+          }).rejects.toThrow('1 operation failed');
+
+          expect(mockExecute).not.toHaveBeenCalledWith('walkIntoMordor', {
```

Comment:
How come execute is not called for this parameter group? Is only the 'guide' parameter group failing parameter validation?

Reply:
oh whoops, this was a copy and paste error. I've updated the test, which helped me understand the bug you were trying to point out above.